src/libcrun/container.c

```diff
@@ -1187,6 +1187,13 @@ container_init_setup (void *args, pid_t own_pid, char *notify_socket, int sync_s
   if (clearenv ())
     return crun_make_error (err, errno, "clearenv");
 
+  // set primary process to 1 explicitly if nothing is configured and LISTEN_FD is not set
+  if (getenv ("LISTEN_PID") == NULL && entrypoint_args->context->preserve_fds == 0)
```

Comment:
it should also check for `LISTEN_FDS` to be set.

If `LISTEN_FDS` is not set, then we don't want to set `LISTEN_PID`

Reply:
the check should be ` entrypoint_args->context->preserve_fds > 0`

Reply:
I concur, that's what runc does:

https://github.com/opencontainers/runc/blob/8b59b768a945babf39e2468dc5d8afa54c511852/utils_linux.go#L279-L282

Reply:
@giuseppe ah I thought `context->preserve_fds` is always `0` when `LISTEN_FDS` is not set but let me confirm.

Reply:
@vrothberg @giuseppe fixed in latest commit.